File: frontend/src/types/share.ts
export interface Share {
  id: number
  userId: number
  photoId: number
  albumId: number
  shareCode: string
  hasPassword: boolean
  expireAt: string
  viewCount: number
  isActive: number
  createdAt: string
  photoTitle: string
  albumName: string
}

export interface ShareCreate {
  photoId?: number
  albumId?: number
  password?: string
  expireHours?: number
}

export interface ShareContent {
  type: string
  content: any
  photos: any[]
}

File: backend/src/main/java/com/photo/entity/Share.java
package com.photo.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("t_share")
public class Share implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long photoId;

    private Long albumId;

    private String shareCode;

    private String password;

    private LocalDateTime expireAt;

    private Integer viewCount;

    private Integer isActive;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
